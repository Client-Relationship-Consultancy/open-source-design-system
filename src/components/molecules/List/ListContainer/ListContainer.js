import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { colourPalette } from "brandColours"

const StyledListContainer = styled.ul`
  border: ${props => (props.border ? `1px solid ${props.theme.black.tint40.hex}` : "none")};
  border-radius: 5px;
  list-style: none;
  padding: 0;
  > li:last-child {
    border-bottom: ${props => (props.endingLine ? "" : "none")};
  }
`
StyledListContainer.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledListContainer.displayName = "StyledListContainer"

const ListContainer = props => {
  const { border, children, endingLine, id, className } = props
  return (
    <StyledListContainer id={id} className={className} border={border} endingLine={endingLine}>
      {children}
    </StyledListContainer>
  )
}
ListContainer.displayName = "ListContainer"
ListContainer.defaultProps = {
  endingLine: true,
}
ListContainer.propTypes = {
  border: PropTypes.bool,
  endingLine: PropTypes.bool,
  children: PropTypes.node,
  id: PropTypes.string,
  className: PropTypes.string,
}

export default ListContainer
