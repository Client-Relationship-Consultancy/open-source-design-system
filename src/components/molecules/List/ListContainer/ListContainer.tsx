import React from "react";
import styled from "styled-components";

import { colourPalette } from "../../../../brandColours";

interface IStyledListContainer {
  border?: boolean;
  endingLine?: boolean;
}

const StyledListContainer = styled.ul<IStyledListContainer>`
  border: ${props => (props.border ? `1px solid ${props.theme.black.tint40.hex}` : "none")};
  border-radius: 5px;
  list-style: none;
  overflow-y: scroll;
  max-height: 25rem;
  padding: 0;
  > li:last-child {
    border-bottom: ${props => (props.endingLine ? "" : "none")};
  }
`;
StyledListContainer.defaultProps = {
  theme: colourPalette.examplePalette,
};
StyledListContainer.displayName = "StyledListContainer";

interface IListContainer extends IStyledListContainer {
  id?: string;
  className?: string;
}
const ListContainer: React.FC<IListContainer> = props => {
  const { border, children, endingLine, id, className } = props;
  const styledListContainerClassName = className ? `ListContainer ${className}` : "ListContainer";
  return (
    <StyledListContainer
      id={id}
      className={styledListContainerClassName}
      border={border}
      endingLine={endingLine}
    >
      {children}
    </StyledListContainer>
  );
};
ListContainer.displayName = "ListContainer";
ListContainer.defaultProps = {
  endingLine: true,
};

export default ListContainer;
