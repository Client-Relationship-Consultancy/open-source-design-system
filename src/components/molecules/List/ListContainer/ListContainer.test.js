import React from "react"
import { mount } from "enzyme"
import ListContainer from "./ListContainer"
import "jest-styled-components"

describe("ListContainer Component Testing", () => {
  const component = mount(
    <ListContainer>
      <li>hello</li>
      <li>world</li>
    </ListContainer>,
  )
  it("Render a <> tag", () => {
    expect(component.find("ul").exists()).toEqual(true)
  })
  it("Default prop of endingLine is true", () => {
    expect(component.prop("endingLine")).toEqual(true)
  })
  it("Border can be toggled", () => {
    expect(component).toHaveStyleRule("border", "none")
    component.setProps({ border: true })
    expect(component).not.toHaveStyleRule("border", "none")
  })
  it("Snapshot Testing", () => {
    expect(component).toMatchSnapshot()
  })
})
