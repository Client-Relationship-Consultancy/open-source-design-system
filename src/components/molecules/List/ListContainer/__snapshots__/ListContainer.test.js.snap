// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer Component Testing Snapshot Testing 1`] = `
.c0 {
  border: 1px solid #C7C7C7;
  border-radius: 5px;
  list-style: none;
  overflow-y: scroll;
  max-height: 25rem;
  padding: 0;
}

<ListContainer
  border={true}
  endingLine={true}
>
  <StyledListContainer
    border={true}
    className="ListContainer"
    endingLine={true}
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
  >
    <StyledComponent
      border={true}
      className="ListContainer"
      endingLine={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "_foldedDefaultProps": Object {
            "theme": Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            },
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "ListContainer__StyledListContainer-sc-1rc26r4-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "border:",
              [Function],
              ";border-radius:5px;list-style:none;overflow-y:scroll;max-height:25rem;padding:0;> li:last-child{border-bottom:",
              [Function],
              ";}",
            ],
          },
          "displayName": "StyledListContainer",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "ListContainer__StyledListContainer-sc-1rc26r4-0",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
    >
      <ul
        className="c0 ListContainer"
      >
        <li>
          hello
        </li>
        <li>
          world
        </li>
      </ul>
    </StyledComponent>
  </StyledListContainer>
</ListContainer>
`;
